src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt

```diff
@@ -91,11 +91,34 @@ data class Order(
         return status != Status.NOT_STARTED
     }
 
+    /**
+     * Makes a copy of the order with the updated fields
+     */
+    fun updateOrder(
+        itemIds: List<String>,
+        callbackUrl: String,
+        orderType: Type,
+        receiver: Receiver
+    ): Order {
+        throwIfInProgress()
+
+        return this.setOrderLines(itemIds)
+            .setCallbackUrl(callbackUrl)
+            .setOrderType(orderType)
+            .setReceiver(receiver)
+    }
+
+    /**
+     * Delete the order as long as it is possible
+     */
+    fun deleteOrder() {
```

Comment:
Wouldn't it be better to call it canDeleteOrder and have it return bool? Then use it in the service to check if order can be deleted?